src/Functions/Public/iaas/Update-vRAvSphereFabricNetwork.ps1

```diff
@@ -0,0 +1,308 @@
+                    ExternalId = $FabricNetworks.externalId
+                    Name = $FabricNetworks.name
+                    Id = $FabricNetworks.id
+                    CreatedAt = $FabricNetworks.createdAt
+                    UpdatedAt = $FabricNetworks.updatedAt
+                    OrganizationId = $FabricNetworks.organizationId
+                    OrgId = $FabricNetworks.orgId
+                    Links = $FabricNetworks._links
+                }
+            
+        }
+
+        function buildUpdateBody() {
+            $DnsServerAddressesJson = ($DnsServerAddresses | ForEach-Object {"`"$_`""}) -join ","
+            $DnsSearchDomainsJson = ($DnsSearchDomains | ForEach-Object {"`"$_`""}) -join ","
+            if($DnsServerAddressesJson -eq "`"`"") {
+                $DnsServerAddressesJson = $null
+            }
+            if($DnsSearchDomainsJson -eq "`"`"") {
+                $DnsSearchDomainsJson = $null
+            }
+
+            # handle Tags
+            $TagsJson = "null"
+            if ($null -ne $Tags) {
+                $TagsJson = ($Tags | ForEach-Object { 
+                    $TagSplit = $_.Split(":")
+                    [PSCustomObject]@{
+                        key = $TagSplit[0]
+                        value = $TagSplit[1]
+                    }   
+                }) | ConvertTo-Json -Depth 5 -Compress
+            }
+
+            # handle items that may not have been supplied but exist already on object
+            # we keep that value just in case so that is not inadvertantly wiped out
+            if ($CurrentRecord.PSObject.Properties.name -match "content") {
+                $CurrentRecord = $CurrentRecord[0]
+            }
+
+            if ($null -eq $Ipv6Cidr) {
+                $Ipv6Cidr = $CurrentRecord.ipv6Cidr
+            }
+            if ($null -eq $IsDefault) {
+                $IsDefault = $CurrentRecord.isDefault
+            }
+            if ($null -eq $IsPublic) {
+                $IsPublic = $CurrentRecord.isPublic
+            }
+            if ($null -eq $Cidr) {
+                $Cidr = $CurrentRecord.cidr
+            }
+
+            $RawJson = @"
+                {
+                    "ipv6Cidr": "$($ipv6)",
+                    "isDefault": $($isDefault | ConvertTo-Json),
+                    "domain": "$($Domain)",
+                    "defaultIpv6Gateway": "$($DefaultIpv6Gateway)",
+                    "dnsServerAddresses": [ $($DnsServerAddressesJson) ],
+                    "isPublic": $($IsPublic | ConvertTo-Json),
+                    "cidr": "$($Cidr)",
+                    "defaultGateway": "$($DefaultGateway)",
+                    "tags": $($TagsJson),
+                    "dnsSearchDomains": [ $($DnsSearchDomainsJson) ]
+                }
+"@
+            $Body = ($RawJson | ConvertFrom-Json) | ConvertTo-Json -Depth 5
+
+            return $Body
+        }
+
+        function buildAccountQuery() {
```

Comment:
Again, I think this function should have parameters to avoid a scoping issue.

Reply:
This is updated in latest commit and tested.